app/controllers/api/v1/authentication_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Api::V1::AuthenticationController < ApplicationController
+    before_action :authorize_request, except: [:login, :create]
+    before_action :get_authorize_user, only: [:retrive]
+    before_action :set_user
+
+    def login
+        if @user and @user&.valid_password?(login_params[:password])
+            render json: { token: JsonWebToken.encode(user_id: @user.id) }, status: :ok
+        else
+            render json: { error: { email: ['El correo o la contraseña no son correctos'] }}, status: :unauthorized
+        end
+    end 
+
+    def create
+        if @user
+            render json: { errors: { email: ['El correo ya tiene una cuenta asociada'] }}
+        else
+            new_user = User.new(user_params)
+            new_user.add_role :admin
+            if new_user.save
+                render json: { token: token = JsonWebToken.encode(user_id: @user.id) }, status: :ok
```

Comment:
Here when you response with the user's token, you are using the blank @user that was set at the beginning of the method execution. Instead of that you have to return the "new_user" variable because there you save all user information.

Reply:
Solved